types/block_finality.go

```diff
@@ -119,14 +120,17 @@ func (b BlockNumberFinality) IsLatest() bool {
 	return b.Block == Latest && b.Offset >= 0
 }
 
-// BlockNumber gets the safe block number from RPC
-func (b *BlockNumberFinality) BlockNumber(ctx context.Context, requester ethereum.ChainReader) (uint64, error) {
+// BlockNumber gets the safe block number and block header from RPC
+func (b *BlockNumberFinality) BlockNumber(
```

Comment:
I would create a new function called `Block` or `Header`, and call it in places where it needs the block itself and not just number.